openff/toolkit/topology/topology.py

```diff
@@ -3002,3 +2181,114 @@ def is_constrained(self, iatom, jatom):
             return self._constrained_atom_pairs[(iatom, jatom)]
         else:
             return False
+
+    def hierarchy_iterator(self, iter_name):
+        """
+        Get a HierarchyElement iterator from all of the molecules in this topology that provide the appropriately
+        named iterator. This iterator will yield HierarchyElements sorted first by the order that molecules are
+        listed in the Topology, and second by the specific sorting of HierarchyElements defined in each molecule.
+
+        Parameters
+        ----------
+        iter_name: string
+            The iterator name associated with the HierarchyScheme to retrieve (for example 'residues' or 'chains')
+
+        Returns
+        -------
+        iterator of HierarchyElement
+        """
+        for molecule in self._molecules:
+            if hasattr(molecule, iter_name):
+                for item in getattr(molecule, iter_name):
+                    yield item
+
+    # DEPRECATED API POINTS
```

Comment:
I'm collecting the API points that no longer make sense, but can be (mostly) harmlessly redirected to new API points with equivalent functionality here.